neuro_san/internals/messages/traced_message.py

```diff
@@ -0,0 +1,103 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+from __future__ import annotations
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+from typing import Union
+
+from langchain_core.messages.base import BaseMessage
+
+
+class TracedMessage(BaseMessage):
+    """
+    Absstract BaseMessage implementation of a message that is to be sent over
+    to be displayed in an observability/tracing service like LangSmith.
+
+    Note that these messages are intended to go to other services
+    so any sensitive information should be redacted via an overridden
+    translate_for_trace() method.
+
+    As of 10/25/25:
+    LangSmith will only show content in its trace viewer even when there are other
+    fields filled in, even in additional kwargs. So this class makes all those fields
+    visible by copying anything displayable into additional_kwargs and nulling out
+    the content field when a trace_source message is provided in the constructor..
+    """
+
+    # Note this is an abstract class so we do not even define the type for it.
+
+    def __init__(self, content: Union[str, List[Union[str, Dict]]] = "",
+                 trace_source: TracedMessage = None,
+                 **kwargs: Any) -> None:
+        """
+        Pass in content as positional arg.
+
+        Args:
+            content: The string contents of the message.
+            trace_source: Another TracedMessage to copy additional_kwargs from.
+            kwargs: Additional fields to pass to the
+        """
+        if trace_source:
+            # If the content is set to something other than None or an empty string,
+            # that is all that LangSmith will ever show.  So put what we want to show
+            # in the additional_kwargs with effectively null content.
+            additional_kwargs: Dict[str, Any] = trace_source.minimal_additional_kwargs()
+            super().__init__(content="", additional_kwargs=additional_kwargs, **kwargs)
```

Comment:
Above: Case for special observability copy constructor. 
Below: normal passthrough constructor to BaseMessage